plugins/out_s3/s3.c

```diff
@@ -981,11 +982,19 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     int size_check = FLB_FALSE;
     int part_num_check = FLB_FALSE;
     int timeout_check = FLB_FALSE;
-    time_t create_time;
     int ret;
     void *payload_buf = NULL;
     size_t payload_size = 0;
     size_t preCompress_size = 0;
+    time_t file_first_log_time = time(NULL);
+
+    /*
+     * When chunk does not exist, file_first_log_time will be 0.
+     * This is only for unit tests. Also, this part is important, please don't remove it.
```

Comment:
But why should it not be removed? You could simplify it by saying, "this prevents unit tests from segfaulting". More description would be good though. 

I actually meant though to put a comment in `cb_s3_flush` that currently our upload functions require chunk to be non-null. To do it around this line: `if ((upload_file != NULL) && (upload_timeout_check == FLB_TRUE || total_file_size_check == FLB_TRUE)) {`

that the `(upload_file != NULL)` should not be removed.